docs/src/content/docs/book/statements.mdx

```diff
@@ -96,6 +98,131 @@ value += 5;     // augmented assignment (one of the many, see below)
 
 :::
 
+## Destructuring assignment
+
+<Badge text="Available since Tact 1.6" variant="tip" size="medium"/><p/>
+
+The destructuring assignment is a concise way to unpack [Structs][s] and [Messages][m] into distinct variables. It mirrors the [instantiation syntax](/book/expressions#instantiation), but instead of creating a new [Struct][s] or [Message][m] it binds every field or some of the fields to their respective variables.
+
+The syntax is derived from the [`let` statement](#let), and instead of specifying the variable name directly it involves specifying the structure type on the left side of the [assignment operator `={:tact}`](/book/operators#assignment), which corresponds to the structure type of the value on the right side.
+
+```tact {6}
+// Definition of Example
+struct Example { number: Int }
+
+fun basic() {
+    // Basic syntax of destructuring assignment (to the left of "="):
+    let Example { number } = Example { number: 42 };
```

Comment:
Maybe we should split these two types of syntax here so that readers won't mistakenly think that these should always be used together? Like make a variable struct and then destructure it.

Reply:
we actually have the variable below, so we can just move its definition above.

Reply:
Or we can even have even more complex expression instead of a variable, like a function call of a function returning a struct